llvm/tools/sycl-post-link/ModuleSplitter.h

```diff
@@ -126,18 +126,43 @@ class ModuleDesc {
              const Properties &Props)
       : M(std::move(M)), EntryPoints(std::move(EntryPoints)), Props(Props) {
     Name = this->EntryPoints.GroupId.str();
+    fixupEntryPointsLinkage();
   }
 
   ModuleDesc(std::unique_ptr<Module> &&M, const std::vector<std::string> &Names,
              StringRef Name = "NoName")
       : M(std::move(M)), Name(Name) {
     rebuildEntryPoints(Names);
+    fixupEntryPointsLinkage();
+  }
+
+  // Some entry points may have linkonce_odr linkage, which does not really work
+  // for an entry point, because either inliner pass or LLVM linker will remove
+  // functions with such linkage, and entry point must be always available by
+  // definition. This function changes linkage of linkonce_odr entry points.
+  // An example of a such entry point is an instantiation of a function template
+  // even marked SYCL_EXTERNAL.
+  // TODO: is it practical to fix FE instead to assign different linkage for
```

Comment:
It will be useful to get @rolandschulz's comment here. Thanks

Reply:
>  either inliner pass or LLVM linker will remove functions with such linkage

They will only remove them if unused. Why do they incorrectly assume they are unused? Is it because you spit into SPMD and SIMD and the entry point into SIMD is unused inside the SIMD part? And therefore it gets removed after splitting?